packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.tsx

```diff
@@ -389,10 +391,22 @@ export const MultiSectionDigitalClock = React.forwardRef(function MultiSectionDi
     ],
   );
 
+  const viewsToRender = React.useMemo(() => {
+    if (!isRtl) {
+      return views;
+    }
+    const digitViews = views.filter((v) => v !== 'meridiem');
+    const result = digitViews.toReversed();
+    if (views.includes('meridiem')) {
+      result.push('meridiem');
+    }
+    return result;
+  }, [isRtl, views]);
+
   const viewTimeOptions = React.useMemo(() => {
     return views.reduce(
-      (result, currentView) => {
-        return { ...result, [currentView]: buildViewProps(currentView) };
+      (res, currentView) => {
+        return { ...res, [currentView]: buildViewProps(currentView) };
```

Comment:
Nitpick: just to avoid unnecessary diff. 🙈 
```suggestion
      (result, currentView) => {
        return { ...result, [currentView]: buildViewProps(currentView) };
```